src/AutoRest.CSharp/Mgmt/Generation/ResourceOperationWriter.cs

```diff
@@ -862,13 +862,10 @@ private void WriteStartLROMethod(CodeWriter writer, RestClientMethod clientMetho
             writer.WriteXmlDocumentationParameter("cancellationToken", "The cancellation token to use.");
             writer.WriteXmlDocumentationRequiredParametersException(nonPathParameters);
 
-            CSharpType? returnType = GetLROReturnType(clientMethod, context);
             CSharpType lroObjectType = clientMethod.Operation.IsLongRunning
                 ? context.Library.GetLongRunningOperation(clientMethod.Operation).Type
                 : context.Library.GetNonLongRunningOperation(clientMethod.Operation).Type;
-            CSharpType responseType = returnType != null ?
-                lroObjectType : typeof(Azure.Operation);
-            responseType = async ? new CSharpType(typeof(Task<>), responseType) : responseType;
+            CSharpType responseType = async ? new CSharpType(typeof(Task<>), lroObjectType) : lroObjectType;
```

Comment:
We have an extension method to do this:
```suggestion
            CSharpType responseType = lroObjectType.WrapAsync(async);
```